llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -855,7 +855,9 @@ InstructionCost VPRegionBlock::cost(ElementCount VF, VPCostContext &Ctx) {
   // For the scalar case, we may not always execute the original predicated
   // block, Thus, scale the block's cost by the probability of executing it.
   if (VF.isScalar())
-    return ThenCost / getPredBlockCostDivisor(Ctx.CostKind);
+    if (auto *VPIRBB = dyn_cast<VPIRBasicBlock>(Then))
```

Comment:
This change doesn't look right, since we know from the code above that `Then` is always a VPBasicBlock. The only way to make this work is to map the recipe back to the original scalar instruction and find it's parent.

Reply:
The code here is dead, we don't compute the scalar cost in VPlan currently, as currently the VPlan for VF = 1 is not 100% accurate representation of the original scalar loop (e.g. could have multiple replicate regions for multiple original instructions in the same predicated block).

The code should be gone after a rebase

Reply:
OK, so just waiting for this PR to rebase then before approving.